tests/package/packagers/test_pandas_packagers.py

```diff
@@ -12,44 +12,146 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+import importlib
 import tempfile
 from pathlib import Path
-from typing import Union
 
 import numpy as np
 import pandas as pd
 import pytest
 
 from mlrun.package.packagers.pandas_packagers import PandasSupportedFormat
 
+# Set up the format requirements dictionary:
+FORMAT_REQUIREMENTS = {
+    PandasSupportedFormat.PARQUET: "pyarrow",
+    PandasSupportedFormat.H5: "tables",
+    PandasSupportedFormat.XLSX: "openpyxl",
+    PandasSupportedFormat.XML: "lxml",
+    PandasSupportedFormat.HTML: "lxml",
+    PandasSupportedFormat.FEATHER: "pyarrow",
+    PandasSupportedFormat.ORC: "pyarrow",
+}
 
-@pytest.mark.parametrize(
-    "obj",
-    [
-        pd.DataFrame(
-            data=np.random.randint(0, 256, (1000, 10)),
-            columns=[f"column_{i}" for i in range(10)],
+
+def check_skipping_pandas_format(fmt: str):
+    if fmt in FORMAT_REQUIREMENTS:
+        try:
+            importlib.import_module(FORMAT_REQUIREMENTS[fmt])
+        except ModuleNotFoundError:
+            return True
```

Comment:
It will be in `sys.modules` only if it was already imported in the same namespace. You can't count on checking the user's (in this case testing's) interpreter environment for availability of a package that way.